### PR TITLE
Removed dependencies to the ValidatedDisplayableProcess.name on BE side + RemoteEnvironment / ModelMigration cleanups

### PR DESCRIPTION
**Pull request by arkadius**
## Describe your changes

RemoteEnvironment / ModelMigration changes:
  * `ProcessMigration.failOnNewValidationError` was removed - it wasn't used anywhere anymore
  * `RemoteEnvironment.testMigration` result types changes
    * `shouldFailOnNewErrors` field was removed - it wasn't used anywhere anymore
    * `converted` field was replaced by the `processName` field which was the only information that was used

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/ProcessTestData.scala [hunk 1]
```diff
@@ -134,12 +138,12 @@ object ProcessTestData {
   val validProcessWithEmptySpelExpr: CanonicalProcess =
     validProcessWithParam("fooProcess", "expression" -> Expression.spel(""))
 
-  val validDisplayableProcess: ValidatedDisplayableProcess = toValidatedDisplayable(validProcess)
+  val validDisplayableProcess: DisplayableProcess = toDisplayable(validProcess)
```
Comment (lciolecki) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/ProcessTestData.scala:141]: Just `displayableProcess`?

**Review by lciolecki — APPROVED** [review]
LGTM

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentInfo.scala [hunk 2]
```diff
@@ -27,7 +27,9 @@ object BuiltInComponentInfo {
   val FragmentInputDefinition: ComponentInfo  = ComponentInfo(ComponentType.BuiltIn, "input")
   val FragmentOutputDefinition: ComponentInfo = ComponentInfo(ComponentType.BuiltIn, "output")
 
+  val AllAvailableForScenario: List[ComponentInfo] = List(Filter, Split, Choice, Variable, RecordVariable)
```
[the remark this answers is not in the document]
Comment (arkadius) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentInfo.scala:30]: Nope, they are also available for fragments. Fragments just had a wider set of available components.

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentInfo.scala [hunk 3]
```diff
@@ -27,7 +27,9 @@ object BuiltInComponentInfo {
...
+
   val All: List[ComponentInfo] =
```
[the remark this answers is not in the document]
Comment (arkadius) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/ComponentInfo.scala:32]: But sometimes I just want to iterate though all built-in components, not only for fragments. I could introduce `AllAvailableForFragments` and have an alias `AllAvailableForFragments = All`, but currently there is no place of usage of  `AllAvailableForFragments` in our code

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/ProcessTestData.scala (hunk 1 above)
Comment (arkadius) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/ProcessTestData.scala:141]: The `valid` prefix in this place and any other in this class is added to show that we prepared valid (correct, without validation errors) scenario. Take a look at `validProcessWithEmptySpelExpr`, `validProcessWithNodeId` etc and `invalidProcess` etc.

**Pull request merged**